### ISSUE TITLE
[Bug]: EfficientAD - Very slow training and bad predictions

### ISSUE THREAD

**Issue by SimonB97** [openvinotoolkit/anomalib#1151]
### Describe the bug

Don't know if it's actually a 'bug', but i'm getting weird results on efficientAD. (I have applied PR#1143 to be able to train on my larger custom dataset.)

Firstly, the training takes ages. Although i'm not using the best GPU in town, a RTX 3050ti, other models train a lot faster. One major aspect to this might be that a batch size of one has to be used. It took about 2 days to train for about 60.000 iterations.
Are you getting similarly slow training speeds?

Second, the actual learning curves are very bad. I guessed the model would perform quite well on my custom data because another student-teacher-based model, reverse distillation with OCBE, performed very well on it. 
Unfortunately, it's very time intensive, with such long training times, to test different hyperparameters to see if it's just some badly chosen parameters (although i used the default ones for the important parameters).

Addtionally, the inference times aren't fast too as stated. about 1.5s/it.

I would be grateful if someone could take a look at my configs to find any misconfigurations or possible issues leading to that behaviour. Below i will share more relevant infos and graphs about the training process.

### Dataset

Folder

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

Model: EfficientAD

not possible to reproduce because of missing custom training data (?)

### OS information

OS information:
OS: win 10
Python version: 3.10.11
Anomalib version: 0.6.0
PyTorch version:
CUDA/cuDNN version: 11.6
GPU models and configuration: rtx3050ti
Any other relevant information: custom data

### Expected behavior

Getting better pixel- (and image-)level auroc scores.

### Screenshots

![image](https://github.com/openvinotoolkit/anomalib/assets/102378134/520e51b6-b867-4e12-ad28-4072860eab9a)

![image](https://github.com/openvinotoolkit/anomalib/assets/102378134/7e448a4c-0935-497e-a2b0-dc4684b16ee3)

![image](https://github.com/openvinotoolkit/anomalib/assets/102378134/5628db93-cc2a-4aba-a187-0647be5dd9d9)

![image](https://github.com/openvinotoolkit/anomalib/assets/102378134/f227cf24-8cd1-49f6-97a9-d3801936b5ba)

![image](https://github.com/openvinotoolkit/anomalib/assets/102378134/efc8a1c1-c4a8-4378-bd8a-88c4a8d4e916)

![image](https://github.com/openvinotoolkit/anomalib/assets/102378134/83177a99-2b70-410e-8cc4-dfac12e3da4f)

![image](https://github.com/openvinotoolkit/anomalib/assets/102378134/548202a2-6ddf-4bdf-aefc-d78f6a393bcb)

![image](https://github.com/openvinotoolkit/anomalib/assets/102378134/486d6311-d5bf-4170-a533-c35e15297839)

### Pip/GitHub

GitHub

### What version/branch did you use?

0.6.0 + pr #1143

### Configuration YAML

```yaml
dataset:
  abnormal_dir: bad
  center_crop: null
  extensions: null
  format: folder
  image_size: 256  # mandatory
  mask: ground_truth
  name: vespa
  normal_dir: good
  normal_test_dir: null
  normalization: imagenet
  num_workers: 8
  path: ./../data/images/data_structure_anomalib
  split_ratio: 0.2
  task: classification
  test_batch_size: 16
  test_split_mode: from_dir
  test_split_ratio: 0.2
  tiling:
    apply: false
    random_tile_count: 16
    remove_border_count: 0
    stride: null
    tile_size: null
    use_random_tiling: false
  train_batch_size: 16
  transform_config:
    eval: c:\Users\benedict\dev\VespaUseCase\machine_learning\src\d01_config\transform_config_1.yaml
    train: null
  val_split_mode: from_test
  val_split_ratio: 0.5
logging:
  log_graph: false
  logger: []
metrics:
  image:
  - F1Score
  - AUROC
  pixel:
  - F1Score
  - AUROC
  threshold:
    manual_image: null
    manual_pixel: null
    method: adaptive
model:
  lr: 0.0001
  model_size: small
  name: efficientad
  normalization_method: min_max
  padding: false
  teacher_out_channels: 384
  weight_decay: 1.0e-05
optimization:
  export_mode: onnx
project:
  path: ./results
  seed: 42
trainer:
  accelerator: auto
  accumulate_grad_batches: 1
  auto_lr_find: false
  auto_scale_batch_size: false
  benchmark: false
  check_val_every_n_epoch: 1
  default_root_dir: null
  detect_anomaly: false
  deterministic: false
  devices: 1
  enable_checkpointing: true
  enable_model_summary: true
  enable_progress_bar: true
  fast_dev_run: false
  gradient_clip_algorithm: norm
  gradient_clip_val: 0
  limit_predict_batches: 1.0
  limit_test_batches: 1.0
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  log_every_n_steps: 50
  max_epochs: 200
  max_steps: -1
  max_time: null
  min_epochs: null
  min_steps: null
  move_metrics_to_cpu: false
  multiple_trainloader_mode: max_size_cycle
  num_nodes: 1
  num_sanity_val_steps: 0
  overfit_batches: 0.0
  plugins: null
  precision: 32
  profiler: null
  reload_dataloaders_every_n_epochs: 0
  replace_sampler_ddp: true
  strategy: null
  sync_batchnorm: false
  track_grad_norm: -1
  val_check_interval: 1.0
visualization:
  image_save_path: null
  log_images: false
  mode: full
  save_images: true
  show_images: false
```

### Transform Config YAML

```
__version__: 1.3.1
transform:
  __class_fullname__: Compose
  additional_targets: {}
  bbox_params: null
  is_check_shapes: true
  keypoint_params: null
  p: 1.0
  transforms:
  - __class_fullname__: Resize
    always_apply: true
    height: 256
    interpolation: 1
    p: 1
    width: 256
  - __class_fullname__: RandomBrightnessContrast
    always_apply: false
    brightness_by_max: true
    brightness_limit:
    - -0.3
    - 0.3
    contrast_limit:
    - -0.3
    - 0.3
    p: 0.5
  - __class_fullname__: RandomSunFlare
    always_apply: false
    angle_lower: 0
    angle_upper: 1
    flare_roi:
    - 0.3
    - 0.4
    - 0.7
    - 0.6
    num_flare_circles_lower: 1
    num_flare_circles_upper: 2
    p: 0.3
    src_color:
    - 255
    - 255
    - 255
    src_radius: 100
  - __class_fullname__: Rotate
    always_apply: false
    border_mode: 4
    crop_border: false
    interpolation: 1
    limit:
    - -10
    - 10
    mask_value: null
    p: 0.5
    rotate_method: largest_box
    value: null
  - __class_fullname__: Affine
    always_apply: false
    cval: 0
    cval_mask: 0
    fit_output: false
    interpolation: 1
    keep_ratio: false
    mask_interpolation: 0
    mode: 0
    p: 0.3
    rotate:
    - 0.0
    - 0.0
    rotate_method: largest_box
    scale:
      x:
      - 1.0
      - 1.0
      y:
      - 1.0
      - 1.0
    shear:
      x: &id001
      - -5
      - 5
      y: *id001
    translate_percent: null
    translate_px:
      x:
      - 0
      - 0
      y:
      - 0
      - 0
  - __class_fullname__: ToTensorV2
    always_apply: true
    p: 1.0
    transpose_mask: false

```

### Logs

```shell
No logs available
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

Edit:
Added tranform config below config

**Comment by alexriedel1** [comment]
You apply transformation during evaluation but not during training, is this intended?

You mention that other methods work better, can you explain this in more detail using the respective metrics?

Also you should not apply normalization to efficientad, this is done internally by the model already. If you provide transforms for eval and not for training, the default transforms for training will be used (that include normalization unless you set normalization: none)

Generally you should not change the default configy values if you are not exactly sure what you're doing I guess.

**Comment by SimonB97** [comment]
> You apply transformation during evaluation but not during training, is this intended?
> 
> You mention that other methods work better, can you explain this in more detail using the respective metrics?
> 
> Also you should not apply normalization to efficientad, this is done internally by the model already. If you provide transforms for eval and not for training, the default transforms for training will be used (that include normalization unless you set normalization: none)
> 
> Generally you should not change the default configy values if you are not exactly sure what you're doing I guess. That also applies to the batch size that you set to train_batch_size: 16

I'm sorry, i didn't provide the correct transform config.. actually i applied transforms only during training, and normalization was turned off. Batch size was 1, too.

I guess the architecture is just not well suited for the domain. 

Because of time constraints i cannot continue right now with working on tuning this model, so no pressure to get this right, but thank you anways!

I'm gonna close the issue and re-open it if necessary.
